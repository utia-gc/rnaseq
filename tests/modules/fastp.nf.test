nextflow_process {

    name "Test Process fastp"
    script "modules/fastp.nf"
    process "fastp"

    test("fastp trims reads -- SE reads") {
        // instantiate helper objects
        def readsLane1 = new ReadsSELane1()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        sampleNumber: "${readsLane1.getSampleNumber()}",
                        lane:         "${readsLane1.getLane()}",
                        readType:     "${readsLane1.getReadType()}",
                        trimStatus:   "raw",
                    ],
                    file("${readsLane1.getR1()}"),
                    file("${readsLane1.getR2()}")
                ]
                input[1] = file("${projectDir}/assets/NO_FILE")
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test trimmed fastqs
            with(process.out.reads) {
                assert size() == 1
                with(get(0)) {
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}_trimmed_R1\.fastq\.gz/
                    assert get(2) ==~ /^.*\/${readsLane1.getStemName()}_trimmed_R2\.NOFILE/
                }
            }
            // test trim log
            with(process.out.log) {
                assert size() == 1
                assert get(0) ==~ /^.*\/${readsLane1.getStemName()}_fastp.json/
            }

            // test trimmed fastq reproducibility
            assert snapshot(process.out.reads).match()
        }

    }

    test("fastp trims reads -- PE reads") {
        // instantiate helper objects
        def readsLane1 = new ReadsPELane1()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        sampleNumber: "${readsLane1.getSampleNumber()}",
                        lane:         "${readsLane1.getLane()}",
                        readType:     "${readsLane1.getReadType()}",
                        trimStatus:   "raw",
                    ],
                        file("${readsLane1.getR1()}"),
                        file("${readsLane1.getR2()}")
                ]
                input[1] = file("${projectDir}/assets/NO_FILE")
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test trimmed fastqs
            with(process.out.reads) {
                assert size() == 1
                with(get(0)) {
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}_trimmed_R1.fastq.gz/
                    assert get(2) ==~ /^.*\/${readsLane1.getStemName()}_trimmed_R2.fastq.gz/
                }
            }
            // test trim log
            with(process.out.log) {
                assert size() == 1
                assert get(0) ==~ /^.*\/${readsLane1.getStemName()}_fastp.json/
            }

            // test trimmed fastq reproducibility
            assert snapshot(process.out.reads).match()
        }

    }


    test("fastp trims reads -- SE reads + adapter fasta") {
        // instantiate helper objects
        def readsLane1 = new ReadsSELane1()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        sampleNumber: "${readsLane1.getSampleNumber()}",
                        lane:         "${readsLane1.getLane()}",
                        readType:     "${readsLane1.getReadType()}",
                        trimStatus:   "raw",
                    ],
                    file("${readsLane1.getR1()}"),
                    file("${readsLane1.getR2()}")
                ]
                input[1] = file("${projectDir}/tests/assets/sample_adapters.fasta")
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test trimmed fastqs
            with(process.out.reads) {
                assert size() == 1
                with(get(0)) {
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}_trimmed_R1\.fastq\.gz/
                    assert get(2) ==~ /^.*\/${readsLane1.getStemName()}_trimmed_R2\.NOFILE/
                }
            }
            // test trim log
            with(process.out.log) {
                assert size() == 1
                assert get(0) ==~ /^.*\/${readsLane1.getStemName()}_fastp.json/
            }

            // test trimmed fastq reproducibility
            assert snapshot(process.out.reads).match()
        }

    }


    test("fastp trims reads -- PE reads + adapter fasta") {
        // instantiate helper objects
        def readsLane1 = new ReadsPELane1()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        sampleNumber: "${readsLane1.getSampleNumber()}",
                        lane:         "${readsLane1.getLane()}",
                        readType:     "${readsLane1.getReadType()}",
                        trimStatus:   "raw",
                    ],
                        file("${readsLane1.getR1()}"),
                        file("${readsLane1.getR2()}")
                ]
                input[1] = file("${projectDir}/tests/assets/sample_adapters.fasta")
                """
            }
        }


        then {
            // test process success
            assert process.success

            // test trimmed fastqs
            with(process.out.reads) {
                assert size() == 1
                with(get(0)) {
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}_trimmed_R1.fastq.gz/
                    assert get(2) ==~ /^.*\/${readsLane1.getStemName()}_trimmed_R2.fastq.gz/
                }
            }
            // test trim log
            with(process.out.log) {
                assert size() == 1
                assert get(0) ==~ /^.*\/${readsLane1.getStemName()}_fastp.json/
            }

            // test trimmed fastq reproducibility
            assert snapshot(process.out.reads).match()
        }

    }

}
