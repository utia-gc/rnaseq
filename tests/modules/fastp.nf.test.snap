{
    "fastp trims reads -- PE reads + adapter fasta": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR6924569",
                        "sampleNumber": "1",
                        "lane": "001",
                        "readType": "paired",
                        "trimStatus": "raw"
                    },
                    "SRR6924569_L001_trimmed_R1.fastq.gz:md5,a26de658644febb1260fd56d2690c01c",
                    "SRR6924569_L001_trimmed_R2.fastq.gz:md5,569eb9ca4d935f290fc1e4a88fa9433d"
                ]
            ]
        ],
        "timestamp": "2024-03-04T12:23:56.438949466"
    },
    "fastp trims reads -- SE reads": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR1066657",
                        "sampleNumber": "3",
                        "lane": "001",
                        "readType": "single",
                        "trimStatus": "raw"
                    },
                    "SRR1066657_L001_trimmed_R1.fastq.gz:md5,fd66b28bfdc8dbc6888912de7bcebd92",
                    "SRR1066657_L001_trimmed_R2.NOFILE:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "timestamp": "2023-10-31T13:44:38.623050842"
    },
    "fastp trims reads -- PE reads": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR6924569",
                        "sampleNumber": "1",
                        "lane": "001",
                        "readType": "paired",
                        "trimStatus": "raw"
                    },
                    "SRR6924569_L001_trimmed_R1.fastq.gz:md5,92338c4ea608209b461a6d747fc272bb",
                    "SRR6924569_L001_trimmed_R2.fastq.gz:md5,ffc0f8521e55d3f76548fc0da8b2f728"
                ]
            ]
        ],
        "timestamp": "2023-10-31T13:44:54.751299753"
    },
    "fastp trims reads -- SE reads + adapter fasta": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR1066657",
                        "sampleNumber": "3",
                        "lane": "001",
                        "readType": "single",
                        "trimStatus": "raw"
                    },
                    "SRR1066657_L001_trimmed_R1.fastq.gz:md5,4259524899c6389b75cf20739402e65c",
                    "SRR1066657_L001_trimmed_R2.NOFILE:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "timestamp": "2024-03-04T12:23:46.481046166"
    }
}